CreeDictionary/API/models.py

```diff
@@ -363,8 +363,8 @@ def fetch_lemma_by_user_query(user_query: str, **kwargs) -> "CreeAndEnglish":
         # utilize the spell relax in descriptive_analyzer
         # TODO: use shared.descriptive_analyzer (HFSTOL) when this bug is fixed:
         # https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/issues/120
-        fst_analyses: Set[str] = set(
-            "".join(a) for a in descriptive_analyzer_foma.analyze(user_query)
+        fst_analyses: Set[ConcatAnalysis] = set(
+            a.concatenate() for a in temp_hfstol.analyze(user_query)
```

Comment:
I could have used `Set[Analysis]` instead of `Set[ConcatAnalysis]` here but that's going to be a big refactor (and hard actually), because in the database, analysis are stored in the format of concatenated analysis. Some lines below, we need to use both the analysis from this set with analysis from the database to construct search results. So for the time being I added a `.concatenate` method to Analysis class I made the code consistently use `ConcatAnalysis`.

Reply:
Fair enough. And I like the name! :)